beetsplug/rewrite.py

```diff
@@ -34,9 +34,7 @@ def rewriter(field, rules):
     def fieldfunc(item):
         value = item._values_fixed[field]
         for pattern, replacement in rules:
-            if pattern.match(value.lower()):
-                # Rewrite activated.
-                return replacement
+            return re.sub(pattern, replacement, value)
```

Comment:
I am afraid you're right, regarding the replacement string example. The example would still be achievable with a `(.*)\s+EP$` search, and `\1` replacement pattern, but I see your point. It's a bit of a hack.
I am not sure what to do here, really. Adding a setting is always an option, of course. To be honest, I was a little surprised to see that the current implementation does not support regex replacement patterns in the first place. Intuitively I figured they should work out of the box, since searching for regexes is possible, too…